### PR TITLE
Remove some usages of CUDA_VERSION macro

### PR DESCRIPTION
**Pull request by snnn**
### Description
We should avoid using the macro since the value of the macro is inaccurate. For example, our prebuilt packages are built with CUDA 11.8 but people may run the binaries with CUDA 11.4.  A runtime function should be used to determine CUDA version.  Like:
```C++
  int cuda_runtime_version = 0;
  CUDA_CALL_THROW(cudaRuntimeGetVersion(&cuda_runtime_version));
  ORT_ENFORCE(cuda_runtime_version >= 11040, "ONNX Runtime needs cuda runtime higher than 11.4");
```

The minimal CUDA version we support is CUDA 11.4.

I didn't touch float16.h. The file needs much more work. I will address it in another PR.

### CODE REVIEW

File: onnxruntime/contrib_ops/cuda/quantization/qordered_ops/qordered_qdq.cc [hunk 1]
```diff
@@ -95,7 +95,7 @@ cublasLtOrder_t GetCublasLtOrderAttr(const OpKernelInfo& info, const char* order
 #endif
 
 QuantizeWithOrder::QuantizeWithOrder(const OpKernelInfo& info) : CudaKernel(info) {
-#if defined(CUDA_VERSION) && CUDA_VERSION >= 11040
+#if defined(USE_CUDA)
```
Comment (tianleiwu) [inline onnxruntime/contrib_ops/cuda/quantization/qordered_ops/qordered_qdq.cc:98]: remove #else below

Comment (snnn) [reply onnxruntime/contrib_ops/cuda/quantization/qordered_ops/qordered_qdq.cc:98]: And they do not need to support ROCM?

Comment (snnn) [reply onnxruntime/contrib_ops/cuda/quantization/qordered_ops/qordered_qdq.cc]: How about the other files in the same directory?

Comment (tianleiwu) [reply onnxruntime/contrib_ops/cuda/quantization/qordered_ops/qordered_qdq.cc]: qordered_ops files are excluded from ROCm so I think it is fine: https://github.com/microsoft/onnxruntime/blob/f0139658316ce231aa5db54b3bc575cd85c6dd4c/cmake/onnxruntime_rocm_hipify.cmake#L58

**Review by tianleiwu — APPROVED** [review]

**Review by tianleiwu — APPROVED** [review]

**Pull request merged**